### PR TITLE
Added files_to_run_first as ordering option.

### PR DESCRIPTION
**Pull request by caiodsc**
Added files to run first as ordering option.

### CODE REVIEW

File: lib/rspec/core/ordering.rb [hunk 1]
```diff
@@ -66,6 +66,20 @@ def order(list)
         end
       end
 
+      class Priority
+        def initialize(files=[])
+          @files = files
+        end
+
+        def order(list)
+          priority_files = list.select do |item|
```
Comment (pirj) [inline lib/rspec/core/ordering.rb:75]: [`partition`](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-partition)?

Comment (caiodsc) [reply lib/rspec/core/ordering.rb:75]: I considered using partition but found it to be too slow in some cases. I will do more research on this.

**Comment by caiodsc** [comment]
I'm writing this PR in response to another approved PR which includes sorting the files by modification date. In my opinion this is not a good heuristic to consider because:
1 - There is actually no correlation between a modification date and the probability of a test to fail. If you changed something in a class and there is a very old integration test that you didn't touch, it will run in just in the end. In other words it's not a good metric.
2 - Also we are losing performance by applying this sort method for the entire test suit. I think it is useful for the recent tests but doesn't make difference for older tests.

I think that the solution for this is to let the user choose which files (in his opinion) are convenient to test first.

Use case:

`bundle exec rspec --priority-files recently_modified1_spec.rb, recently_modified2_spec.rb`

In other words, we put the responsability on the user, if he wants he can sort the files on his side by whatever he wants(modification date, size of file, etc) and pass the priority files as arguments.

Final use case (using along with github actions):

```
      run: |
        git fetch origin master:master
        CHANGED_FILES=`git diff master --name-only | grep '_spec.rb$'`
        CHANGED_FILES=`${CHANGED_FILES//$'\n'/\,}`
        bundle exec rspec --priority-files $CHANGED_FILES
```

In this example we are testing the spec files that changed in a specific pull request first (by setting them as priority files). What do you think?

File: lib/rspec/core/ordering.rb (hunk 1 above)
Comment (caiodsc) [reply lib/rspec/core/ordering.rb]: Done!

**Comment by JonRowe** [comment]
:wave: You can already do this by directly passing files and using `--order defined`.
